downstairs/src/extent_inner_sqlite.rs

```diff
@@ -342,12 +342,37 @@ impl SqliteMoreInner {
                 (job_id.0, self.extent_number, n_contiguous_blocks as u64)
             });
 
-            nix::sys::uio::pread(
-                self.file.as_fd(),
-                &mut buf,
-                first_req.offset.value as i64 * block_size as i64,
-            )
-            .map_err(|e| CrucibleError::IoError(e.to_string()))?;
+            // SAFETY: the buffer has sufficient capacity, and this is a valid
+            // file descriptor.
+            let r = unsafe {
+                libc::pread(
+                    self.file.as_raw_fd(),
+                    buf.spare_capacity_mut().as_mut_ptr().cast(),
```

Comment:
similarly, IMHO, `ptr::cast()` without a type annotation obscures the type being cast to...